MvvmCross/Core/Binding/Bindings/MvxFullBinding.cs

```diff
@@ -94,12 +107,14 @@ private void UpdateTargetOnBind()
         {
             if (this.NeedToUpdateTargetOnBind && this._sourceStep != null)
             {
-                // note that we expect Bind to be called on the UI thread - so no need to use RunOnUIThread here
+                cancelSource.Cancel();
+                cancelSource = new CancellationTokenSource();
```

Comment:
Well no:  if you dispose it here, the current subscribers of cancelSource will crash (they are async and may be doing other work before retesting the cancelSource.Token variable they have captured using closures).